### ISSUE TITLE
Packages for 0.4

### ISSUE THREAD

**Issue by simonbyrne** [JuliaLang/JuliaBox#315]
Would it be possible to get the same set of packages set up for 0.3 to work for 0.4, now that it has been released?

**Comment by mlubin** [comment]
:+1:

**Comment by ViralBShah** [comment]
The pre-installed packages has created a lot of problems for maintenance. The earlier architecture did not allow for users to install their own packages. That has now been fixed, and the plan going forward is for users to maintain their own packages.

We have been going through a lot of package issues on JuliaBox in the last few weeks. Do see the recent issues.

Open to new ideas to help making this better.

**Comment by mlubin** [comment]
@ViralBShah, for tutorials it's a huge boon to have a lot of the common packages already there and ready to use. Especially the optimization packages, which have large binary dependencies. We can't ask people to run ``Pkg.add()`` in a tutorial and wait 15 minutes for compilation to finish.

**Comment by simonbyrne** [comment]
I'm also getting a lot of precompilation errors: e.g. I can't get either PyPlot, Winston or Gadlfy to work.

**Comment by mlubin** [comment]
Yes, doing ``Pkg.build("Cbc")`` on a clean JuliaBox session is broken: https://github.com/JuliaLang/julia/issues/12508#issuecomment-149375414

**Comment by mlubin** [comment]
Not to mention that the first ``using Gadfly`` takes minutes to run. I'm preparing a tutorial for this week and I can't say that Julia 0.4 on JuliaBox is in a presentable state, I'll be directing users to 0.3.

**Comment by tanmaykm** [comment]
JuliaBox (with some modifications) can support multiple "package bundles". Package bundles are basically tar-gz of `.julia` folder, something like what is created by https://github.com/JuliaLang/JuliaBox/blob/master/container/interactive/setup_julia.sh. We can have anyone create and publish them on JuliaBox. Users can select the package bundle they need for their session.

This model can probably be tried out externally first, by building and putting up such bundles for download. Users can `wget` them and add them to the `LOAD_PATH`.

**Comment by mlubin** [comment]
@tanmaykm, will that work with precompilation also?

**Comment by tanmaykm** [comment]
I think precompilation uses `Pkg.dir()` only for the cache. So it will probably compile packages loaded from LOAD_PATH again on first use.

**Comment by ViralBShah** [comment]
Can we try this out and confirm back to everyone? I think it is important to have a good experience for folks doing tutorials and such, and if package bundles work, that could be a good solution.

In the meanwhile, @mlubin it would be best to stay with 0.3.

**Comment by tanmaykm** [comment]
This is exactly how packages are being loaded for 0.3 now.

**Comment by simonbyrne** [comment]
The crux of the problem seems to be the [Compat already loaded issue](https://github.com/JuliaLang/julia/issues/12508#issuecomment-149410764).

As it stands, JuliaBox is essentially unusable with 0.4.

**Comment by tanmaykm** [comment]
created #319 to discuss on this topic further, please put up your ideas.
